### PR TITLE
20240226 release notes

### PR DESCRIPTION
**Pull request by shashankbarsin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+  * Label `kubernetes.azure.com/managedby: aks` has been introduced to all managed addon components on cluster.
+
+* Bug fixes
+  * [Pod overhead](https://kubernetes.io/docs/concepts/containers/runtime-class/#pod-overhead) of memory 2Gi added to `kata-cc-isolation` [RuntimeClass](https://kubernetes.io/docs/concepts/containers/runtime-class/) to address issue where too many pods being created to use too much of the node's memory was resulting in random processes being OOM killed.
+  * Fixed issue that was causing PUT operations on AKS clusters that were using [Bring your own Container Network Interface (CNI) plugin](https://learn.microsoft.com/azure/aks/use-byo-cni?tabs=azure-cli) to fail when the request didn't contain the `networkProfile.podCIDR` property.
+  * In AKS clusters of version >= 1.27.0, fixed a race condition in the iptables mode of kube-proxy that could result in some updates getting lost (for example, when a service gets a new endpoint).
+  * Fixed a race condition that could cause [upgrade from kubenet to Azure CNI Overlay](https://learn.microsoft.com/azure/aks/azure-cni-overlay?tabs=kubectl#kubenet-cluster-upgrade) to fail.
+
+
```
Comment (allyford) [inline CHANGELOG.md:28]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,45 @@
...
+  * Upgraded `kappie-agent` Linux and Windows images used in [AKS Network Observability](https://learn.microsoft.com/azure/aks/network-observability-overview) to v0.1.4 and v0.1.3 respectively.
+  * Upgraded ACI provider for the Virtual Kubelet to [v1.6.1](https://github.com/virtual-kubelet/azure-aci/releases/tag/v1.6.1)
+  * Cilium version has been updated to [1.14.4](https://github.com/cilium/cilium/releases/tag/v1.14.4) for AKS clusters with kubernetes versions >= 1.29.0.
+  * Azure Linux image has been updated to [Azure Linux - 202402.12.0](vhd-notes/AzureLinux/202402.12.0.txt).
+  * AKS Ubuntu 22.04 image has been updated to [AKSUbuntu-2204-202402.12.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202402.12.0.txt).
+  * Azure Windows 2019 Image has been updated to [Azure Windows 2019 - 17763.5458.240218](vhd-notes/AKSWindows/2019/17763.5458.240218.txt).
```
Comment (allyford) [inline CHANGELOG.md:40]:
```suggestion
  * Windows Server 2019 Image has been updated to [Windows Server 2019 - 17763.5458.240218](vhd-notes/AKSWindows/2019/17763.5458.240218.txt).
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,45 @@
...
+  * Azure Windows 2022 Image has been updated to [Azure Windows 2022 - 20348.2322.240218](vhd-notes/AKSWindows/2022/20348.2322.240218.txt).
```
Comment (allyford) [inline CHANGELOG.md:41]:
```suggestion
  * Windows Server 2022 Image has been updated to [Windows Server 2022 - 20348.2322.240218](vhd-notes/AKSWindows/2022/20348.2322.240218.txt).
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,45 @@
...
+* Beginning with AKS support of Kubernetes 1.29 in preview, optimized reservation logic reduces Kube-reserved memory by up to 20% depending on the node configuration and will apply to all clusters. More information can be found [here](https://learn.microsoft.com/azure/aks/concepts-clusters-workloads#memory)
+
+### Release notes
+
+* Preview features
+  * Istio revision asm-1-20 is now available with Istio-based service mesh add-on. More information on performing canary upgrade for the new minor revision of Istio can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade). Istio revision asm-1-18 is no longer supported.
+  * PodIPAllocationMode property introduced on node pools for static block allocation in the case of Azure CNI.
```
Comment (allyford) [inline CHANGELOG.md:16]: Docs? and is this a preview feature?

Comment (shashankbarsin) [reply CHANGELOG.md:16]: yes under preview. waiting for docs link from Chase

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,44 @@
...
+
+* Behavioral change
+  * `ignoreUnfixed` is not set to `false` in [scanner options](https://eraser-dev.github.io/eraser/docs/customization#scanner-options) for [Image Cleaner](https://learn.microsoft.com/azure/aks/image-cleaner) so that images with vulnerabilities are deleted even if there is no fix/patch available for it yet.
```
Comment (wangyira) [inline CHANGELOG.md:19]: Is this a bug fix?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,44 @@
...
+* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* Beginning with AKS support of Kubernetes 1.29 in preview, optimized reservation logic reduces Kube-reserved memory by up to 20% depending on the node configuration and will apply to all clusters. More information can be found [here](https://learn.microsoft.com/azure/aks/concepts-clusters-workloads#memory)
```
Comment (palma21) [inline CHANGELOG.md:10]: 1.29 is out, ensure these are in the version changes table in the docs and can remove

@kaarthis cc

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,44 @@
...
+
+### Release notes
+
+* Preview features
+  * Istio revision asm-1-20 is now available with Istio-based service mesh add-on. More information on performing canary upgrade for the new minor revision of Istio can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade). Istio revision asm-1-18 is no longer supported.
```
Comment (palma21) [inline CHANGELOG.md:15]: Component update?

File: CHANGELOG.md (hunk 5 above)
Comment (palma21) [reply CHANGELOG.md:19]: Not or now?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,44 @@
...
+  * PodIPAllocationMode property introduced on node pools for static block allocation in the case of Azure CNI.
+
+* Behavioral change
+  * `ignoreUnfixed` is not set to `false` in [scanner options](https://eraser-dev.github.io/eraser/docs/customization#scanner-options) for [Image Cleaner](https://learn.microsoft.com/azure/aks/image-cleaner) so that images with vulnerabilities are deleted even if there is no fix/patch available for it yet.
+  * Label `kubernetes.azure.com/managedby: aks` has been introduced to all managed addon components on cluster.
```
Comment (palma21) [inline CHANGELOG.md:20]: Link and close respective issue in the roadmao

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,44 @@
...
+  * Fixed issue that was causing PUT operations on AKS clusters that were using [Bring your own Container Network Interface (CNI) plugin](https://learn.microsoft.com/azure/aks/use-byo-cni?tabs=azure-cli) to fail when the request didn't contain the `networkProfile.podCIDR` property.
+  * In AKS clusters of version >= 1.27.0, fixed a race condition in the iptables mode of kube-proxy that could result in some updates getting lost (for example, when a service gets a new endpoint).
+  * Fixed a race condition that could cause [upgrade from kubenet to Azure CNI Overlay](https://learn.microsoft.com/azure/aks/azure-cni-overlay?tabs=kubectl#kubenet-cluster-upgrade) to fail.
+
+* Component updates
+  * Open Service Mesh upgraded to v1.2.8 with Envoy upgraded to v1.26.7 to address vulnerabilities CVE-2024-23324, CVE-2024-23325, CVE-2024-23322, CVE-2024-23323, and CVE-2024-23327.
+  * For Node Auto Provisioning, Karpenter is upgraded to [v0.33.0](https://github.com/kubernetes-sigs/karpenter/releases/tag/v0.33.0) and its Azure provider is upgraded to [v0.3.0](https://github.com/Azure/karpenter-provider-azure/releases/tag/v0.3.0).
```
Comment (palma21) [inline CHANGELOG.md:30]: No updates worth noting on preview features?

File: CHANGELOG.md (hunk 5 above)
Comment (shashankbarsin) [reply CHANGELOG.md:19]: good catch. now :)

Comment (shashankbarsin) [reply CHANGELOG.md:19]: @wangyira - not a bug IMO, just changing default towards best practices IMO

File: CHANGELOG.md (hunk 8 above)
Comment (shashankbarsin) [reply CHANGELOG.md:20]: done

File: CHANGELOG.md (hunk 9 above)
Comment (shashankbarsin) [reply CHANGELOG.md:30]: Added KAITO after confirming

**Review by palma21 — APPROVED** [review]
Please confirm pod subnet lrb

**Review by kaarthis — CHANGES_REQUESTED** [review]
Could we pls add the patches.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,47 @@
...
+* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes is cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
```
[the remark this answers is not in the document]
Comment (shashankbarsin) [reply CHANGELOG.md:9]: Added

**Review by KidiIT — COMMENTED** [review]
All reviewed and approved🫶❤️

**Pull request merged**